code/game/objects/items/tanks/tanks.dm

```diff
@@ -300,3 +309,160 @@
 
 	else if(integrity < 3)
 		integrity++
+
+// ---------- Procs below are for tanks that are used exclusively in 1-tank bombs ----------
+
+/obj/item/tank/update_icon()
+	cut_overlays()
+	if(tank_assembly)
+		add_overlay(tank_assembly.icon_state)
+		copy_overlays(tank_assembly)
+		add_overlay("bomb_assembly")
+
+/obj/item/tank/wrench_act(mob/living/user, obj/item/I)
+	to_chat(user, span_notice("You disassemble the bomb!"))
+	if(tank_assembly)
+		tank_assembly.forceMove(drop_location())
+		tank_assembly.master = null
+		tank_assembly = null
+	throw_speed = initial(throw_speed)
+	status = FALSE
+	update_icon()
+	return TRUE
+
+/obj/item/tank/welder_act(mob/living/user, obj/item/I)
+	. = FALSE
+	if(status)
+		to_chat(user, span_notice("[src] already has a pressure hole!"))
+		return
+	if(!I.tool_start_check(user, amount=0))
+		return
+	if(I.use_tool(src, user, 0, volume=40))
+		status = TRUE
+		log_bomber(user, "welded a single tank bomb,", src, "| Temp: [air_contents.return_temperature()-T0C]")
+		to_chat(user, span_notice("A pressure hole has been bored to [src]'s valve. \The [src] can now be ignited."))
+		add_fingerprint(user)
+		return TRUE
+
+/obj/item/tank/receive_signal()	//This is mainly called by the sensor through sense() to the holder, and from the holder to here.
+	audible_message("[icon2html(src, hearers(src))] *beep* *beep* *beep*")
+	playsound(src, 'sound/machines/triple_beep.ogg', ASSEMBLY_BEEP_VOLUME, TRUE)
+	sleep(1 SECONDS)
+	if(QDELETED(src))
+		return
+	if(status)
+		ignite()	//if its not a dud, boom (or not boom if you made shitty mix) the ignite proc is below, in this file
+	else
+		release()
+
+//Assembly / attached device memes
+
+/obj/item/tank/Crossed(atom/movable/AM) //for mousetraps
+	..()
+	if(tank_assembly)
+		tank_assembly.Crossed(AM)
```

Comment:
All of these can be signals that ``tank_assembly`` will register onto the tank.

Reply:
I'll see what I can do

Reply:
Having some issues getting it to work with signals (also there's no signal for on_found apparently) so hopefully it's not too big of an issue to keep it the way it is